src/cli_utils.rs

```diff
@@ -118,3 +118,48 @@ pub fn print_language<W>(sink: &mut W,
 pub fn print_inaccuracy_warning<W>(sink: &mut W) -> io::Result<()> where W: Write {
     writeln!(sink, "Note: results can be inaccurate for languages marked with '{}'", IDENT_INACCURATE)
 }
+
+// This is for a best efforts approach where we use box drawing characters
+// for the separator lines, but only if we can easily determine that
+// the terminal encoding is set to UTF8. We use the POSIX environment variables for that.
+//
+// The environment variables are not set by default under windows except in posix-y
+// environments like WSL, cygwin, MSYS2.
+// Unicode is badly supported in cmd.exe and powershell and not activated by default
+// so we just fall back on ASCII there without checking the so called code page.
+//
+// The variables are assumed to follow XGP syntax:
+// `language[_territory[.codeset]][@modifier]`
+// https://www.gnu.org/software/libc/manual/html_node/Locale-Names.html
+pub fn locale_requests_utf8() -> bool {
+    // LC_ALL overrides LC_CTYPE which overrides LANG
+    // LC_CTYPE is the locale category used for the same purpose in the `tree` utility
+    let env_variables = ["LC_ALL", "LC_CTYPE", "LANG"];
+    let env = env_variables.iter()
+        .map(|&env| std::env::var(env))
+        .find(|env|
+            // filter out unset and empty variables
+            env != &Err(std::env::VarError::NotPresent)
+                && env != &Ok("".to_owned())
```

Comment:
This should be `&Ok(String::new())` instead of `&Ok("".to_owned())` for clarity.